renderdoc/driver/vulkan/wrappers/vk_cmd_funcs.cpp

```diff
@@ -7570,6 +7571,379 @@ void WrappedVulkan::vkCmdEndRendering(VkCommandBuffer commandBuffer)
+
+    byte *memory = GetTempMemory(tempmemSize);
+    auto *unwrappedInfos = reinterpret_cast<VkAccelerationStructureBuildGeometryInfoKHR *>(memory);
+    memory += sizeof(VkAccelerationStructureBuildGeometryInfoKHR) * infoCount;
+
+    for(uint32_t i = 0; i < infoCount; ++i)
+      unwrappedInfos[i] = *UnwrapStructAndChain(m_State, memory, &pInfos[i]);
+
+    ObjDisp(commandBuffer)
+        ->CmdBuildAccelerationStructuresIndirectKHR(Unwrap(commandBuffer), infoCount,
+                                                    unwrappedInfos, pIndirectDeviceAddresses,
+                                                    pIndirectStrides, ppMaxPrimitiveCounts);
+  }
+
+  return true;
+}
+
+void WrappedVulkan::vkCmdBuildAccelerationStructuresIndirectKHR(
+    VkCommandBuffer commandBuffer, uint32_t infoCount,
+    const VkAccelerationStructureBuildGeometryInfoKHR *pInfos,
+    const VkDeviceAddress *pIndirectDeviceAddresses, const uint32_t *pIndirectStrides,
+    const uint32_t *const *ppMaxPrimitiveCounts)
+{
+  {
+    size_t tempmemSize = sizeof(VkAccelerationStructureBuildGeometryInfoKHR) * infoCount;
+    for(uint32_t i = 0; i < infoCount; ++i)
+      tempmemSize += GetNextPatchSize(&pInfos[i]);
+
+    byte *memory = GetTempMemory(tempmemSize);
+    auto *unwrappedInfos = reinterpret_cast<VkAccelerationStructureBuildGeometryInfoKHR *>(memory);
+    memory += sizeof(VkAccelerationStructureBuildGeometryInfoKHR) * infoCount;
+
+    for(uint32_t i = 0; i < infoCount; ++i)
+      unwrappedInfos[i] = *UnwrapStructAndChain(m_State, memory, &pInfos[i]);
+
+    SERIALISE_TIME_CALL(ObjDisp(commandBuffer)
+                            ->CmdBuildAccelerationStructuresIndirectKHR(
+                                Unwrap(commandBuffer), infoCount, unwrappedInfos,
+                                pIndirectDeviceAddresses, pIndirectStrides, ppMaxPrimitiveCounts));
+  }
+
+  if(IsActiveCapturing(m_State))
+  {
+    {
+      CACHE_THREAD_SERIALISER();
+
+      SCOPED_SERIALISE_CHUNK(VulkanChunk::vkCmdBuildAccelerationStructuresIndirectKHR);
+      Serialise_vkCmdBuildAccelerationStructuresIndirectKHR(ser, commandBuffer, infoCount, pInfos,
+                                                            pIndirectDeviceAddresses,
+                                                            pIndirectStrides, ppMaxPrimitiveCounts);
+
+      m_FrameCaptureRecord->AddChunk(scope.Get());
+    }
+
+    for(uint32_t i = 0; i < infoCount; ++i)
+    {
+      const VkAccelerationStructureBuildGeometryInfoKHR &geomInfo = pInfos[i];
+      if(geomInfo.srcAccelerationStructure != VK_NULL_HANDLE)
+        GetResourceManager()->MarkResourceFrameReferenced(
+            GetResID(geomInfo.srcAccelerationStructure), eFrameRef_Read);
+
+      FrameRefType refType = pInfos[i].mode == VK_BUILD_ACCELERATION_STRUCTURE_MODE_BUILD_KHR
+                                 ? eFrameRef_CompleteWrite
+                                 : eFrameRef_PartialWrite;
```

Comment:
This reftype can actually be completewrite either way, since the source acceleration structure does not semantically impact the build step and can (and will) be omitted.